app/uk/gov/hmrc/soletraderidentificationfrontend/controllers/CannotConfirmBusinessErrorController.scala

```diff
@@ -64,13 +64,18 @@ class CannotConfirmBusinessErrorController @Inject()(mcc: MessagesControllerComp
                   formAction = routes.CannotConfirmBusinessErrorController.submit(journeyId),
                   form = formWithErrors
                 ))
-              , {
-                tryAgain =>
-                  if (tryAgain) Redirect(journeyConfig.continueUrl + s"?journeyId=$journeyId")
-                  else Redirect(routes.CaptureFullNameController.show(journeyId))
-              }
+              , tryAgain =>
+                if (tryAgain) {
+                  submissionService.submit(journeyId)
+                  Redirect(journeyConfig.continueUrl + s"?journeyId=$journeyId")
+                } else {
+                  soleTraderIdentificationService.removeAllData(journeyId)
+                  Redirect(routes.CaptureFullNameController.show(journeyId).url)
+                }
```

Comment:
We're not waiting for this removal of data to finish before redirecting back. I think it would be safer if we did, so something like this:
`soleTraderIdentificationService.removeAllData(journeyId).map {
_ => Redirect()
}`